### PR TITLE
Create session model

### PR DESCRIPTION
**Pull request by benclmnt**

### CODE REVIEW

**Comment by benclmnt** [comment]
@dhafinrazaq @maguireong can help to look at it?

@tanweijie123 will this change be compatible with the change you proposed for #68 ?

File: src/main/java/seedu/address/model/session/Interval.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+/**
+ * Represents a Session's interval in FitEgo.
+ * Guarantees: immutable; is valid as declared in {@link #isValidInterval(int)}
+ */
+public class Interval {
+    public static final String MESSAGE_CONSTRAINTS = "Intervals can start at any time, and it have a positive duration";
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("E, dd MMM yyyy, HH:mm");
+
+    public final LocalDateTime start;
```
Comment (dhafinrazaq) [inline src/main/java/seedu/address/model/session/Interval.java]: Should this be non-public?
Screen Shot 2020-09-29 at 5 58 08 PM

**Review by dhafinrazaq — COMMENTED** [review]
the design looks good & similar to the client. however, should the class fields be non-public? (according to the screenshot provided). I'm also not sure since AB3's implementation on the class fields is also public final.

**Comment by tanweijie123** [comment]
> @tanweijie123 will this change be compatible with the change you proposed for #68 ?

Yes. All the test case refactoring can be used. (because I have not started with that) 
Gym, Interval, ExerciseType can be imported because currently, I am using it as String type 
We can merge the Session to accommodate with ^. 

Can refer to #70

File: src/main/java/seedu/address/model/session/Interval.java (hunk 1 above)
Comment (tanweijie123) [reply src/main/java/seedu/address/model/session/Interval.java]: My DateTime formatter also different from yours
DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm")

Comment (benclmnt) [reply src/main/java/seedu/address/model/session/Interval.java]: Oh right, I'll make it private

**Review by dhafinrazaq — APPROVED** [review]
lgtm

**Pull request merged**